content/en/docs/concepts/configuration/secret.md

```diff
@@ -842,9 +842,9 @@ a certificate and its associated key that are typically used for TLS.
 One common use for TLS secrets is to configure encryption in transit for
 an [Ingress](/docs/concepts/services-networking/ingress/), but you can also use it
 with other resources or directly in your workload.
-When using this type of Secret, the `tls.key` and the `tls.crt` key must be provided
-in the `data` (or `stringData`) field of the Secret configuration, although the API
-server doesn't actually validate the values for each key.
+When using this type of Secret, the `tls.key` and the `tls.crt` key must be base64
+encoded and provided in the `data` (or `stringData`) field of the Secret configuration,
```

Comment:
If you set `stringData` I think you don't need the extra base64 encoding. Does that sound right?

Reply:
That's supposed to be the case, but when I use `stringData`, it still somehow changes back to `data` on the cluster.
I am using k8s version 1.26.4 and not sure if it's an issue on my end. Are you able to reproduce this?

Reply:
That's how all Secrets work: you can write either `data` or `stringData`, but the canonical form that the Kubernetes API returns is to have the data base64 encoded in the `data` field.

Right now, https://kubernetes.io/docs/concepts/configuration/secret/ doesn't explain this well.

Reply:
Oh, thanks.
However, the challenge now is that the values for both `tls.crt` and `tls.key` for either `data` or `stringData` remain the same. And the workload(nginx daemonset in my case) keeps trying to decode the value and giving me the error 
```
failed to find any PEM data in certificate input
```
Which isn't the case when I provide base64 encoded values.

Reply:
If you provide a `stringData` input that represents the PEM-encoded certificate and private key, things should work. Reading back the Secret reveals the that `data` is set to include the PEM-encoded certificate and private key, both base64 encoded.

If you provide a `data` input that represents the PEM-encoded certificate and private key (base64 encoded), things should work. Reading back the Secret reveals the that `data` is set to include the PEM-encoded certificate and private key, both base64 encoded.

I hope that makes things slightly more clear to you, and helps you write an explanation that helps all readers.
---

PS To make things harder to follow, PEM encoding uses base64 as part of its implementation. However I would largely ignore that detail and treat the PEM data as opaque.

Reply:
Thanks. It does.
Let me add a corresponding explanation in the PR

Reply:
Thanks. It does.
Let me add a corresponding explanation in the PR